glean-core/src/database/mod.rs

```diff
@@ -147,6 +165,59 @@ impl Database {
         }
     }
 
+    /// Checks if the 'user' lifetime metric is expired and, if so, removes it.
+    ///
+    /// ## Arguments
+    ///
+    /// * `storage_name`: The storage name to iterate over.
+    /// * `metric_id_slice`: The metric id to check.
+    /// ## Return value
+    ///
+    /// Returns a `true` if the metric is expired and should not be reported,
+    /// `false` otherwise.
+    ///
+    /// ## Panics
+    ///
+    /// This function will **not** panic on database errors.
+    fn check_and_remove_expired_user_lifetime(
+        &self,
+        storage_name: &str,
+        metric_id_slice: &[u8],
```

Comment:
Why the `slice` in the name?

Reply:
Mh, that's a good question. I feel like a coward, but that's how I got the Rust compiler to stop screaming at me. Otherwise, it was complaining about `&str` not being allowed when attempting to look up items in the map.